presto-spi/src/main/java/com/facebook/presto/spi/predicate/TupleDomain.java

```diff
@@ -387,11 +387,20 @@ else if (isNone()) {
         }
 
         Map<T, Domain> simplified = domains.get().entrySet().stream()
-                .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().simplify()));
+                .collect(toLinkedMap(Map.Entry::getKey, e -> e.getValue().simplify()));
 
         return TupleDomain.withColumnDomains(simplified);
     }
 
+    private static <T, K, U> Collector<T, ?, Map<K, U>> toLinkedMap(Function<? super T, ? extends K> keyMapper, Function<? super T, ? extends U> valueMapper)
+    {
+        return toMap(
+                keyMapper,
+                valueMapper,
+                (u, v) -> { throw new IllegalStateException(String.format("Duplicate key %s", u)); },
```

Comment:
static import `format`

Reply:
Guava throws `IllegalArgumentException` in case of a conflict. Let's follow the Guava's convention.

Reply:
Also it makes sense to print the conflicting key, in case it ever fails. e.g.: https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/ImmutableMap.java#L215

Reply:
As discussed, Guava is not permissible in presto-spi,

Updated the change list as discussed - full cherry-pick.